src/libraries/Microsoft.PowerFx.Core/IR/Nodes/BinaryOpKind.cs

```diff
@@ -78,5 +80,16 @@ internal enum BinaryOpKind
         DynamicGetField,
 
         // And, Or, Pow, Concatenate get represented as FunctionNodes with lambdas to handle short-circuiting
+        // Included here to make the matrix cleaner, should not be generated in IR.
```

Comment:
Does this seem reasonable, or should we implement a separate enum that covers the matrix, and then convert it to the binaryopkind used in the IR (like we do for CoercionKind->UnaryOpKind)